File: chain-sdk/clojure/src/on_chain_api/specs/mint_v3_dto.clj
(ns on-chain-api.specs.mint-v3-dto
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            )
  (:import (java.io File)))


(def mint-v3-dto-data
  {
   (ds/opt :entry_time) inst?
   (ds/opt :recv_time) inst?
   (ds/opt :block_number) int?
   (ds/opt :id) string?
   (ds/opt :transaction) string?
   (ds/opt :timestamp) string?
   (ds/opt :pool) string?
   (ds/opt :token_0) string?
   (ds/opt :token_1) string?
   (ds/opt :owner) string?
   (ds/opt :sender) string?
   (ds/opt :origin) string?
   (ds/opt :amount) string?
   (ds/opt :amount_0) string?
   (ds/opt :amount_1) string?
   (ds/opt :amount_usd) string?
   (ds/opt :tick_lower) string?
   (ds/opt :tick_upper) string?
   (ds/opt :log_index) string?
   (ds/opt :vid) int?
   })

(def mint-v3-dto-spec
  (ds/spec
    {:name ::mint-v3-dto
     :spec mint-v3-dto-data}))

File: chain-sdk/javascript-closure-angular/API/Client/MintV3DTO.js
goog.provide('API.Client.MintV3DTO');

/**
 * Mint entities are created for every emitted Mint event on the Uniswap core contracts. The Mint entity stores key data about the event like token amounts, who sent the transaction, and more.
 * @record
 */
API.Client.MintV3DTO = function() {}

/**
 * @type {!Date}
 * @export
 */
API.Client.MintV3DTO.prototype.entryTime;

/**
 * @type {!Date}
 * @export
 */
API.Client.MintV3DTO.prototype.recvTime;

/**
 * Number of block in which entity was recorded.
 * @type {!number}
 * @export
 */
API.Client.MintV3DTO.prototype.blockNumber;

/**
 * Transaction hash + '#' + index in mints Transaction array.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.id;

/**
 * Which txn the mint was included in.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.transaction;

/**
 * Time of transaction.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.timestamp;

/**
 * Pool address.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.pool;

/**
 * Reference to token0 as stored in pool contract.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.token0;

/**
 * Reference to token1 as stored in pool contract.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.token1;

/**
 * Owner of position where liquidity minted to.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.owner;

/**
 * The address that minted the liquidity.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.sender;

/**
 * Transaction origin: the EOA (Externally Owned Account) that initiated the transaction.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.origin;

/**
 * Amount of liquidity minted.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.amount;

/**
 * Amount of token 0 minted.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.amount0;

/**
 * Amount of token 1 minted.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.amount1;

/**
 * Derived amount based on available prices of tokens.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.amountUsd;

/**
 * Lower tick of the position.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.tickLower;

/**
 * Upper tick of the position.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.tickUpper;

/**
 * Order within the transaction.
 * @type {!string}
 * @export
 */
API.Client.MintV3DTO.prototype.logIndex;

/**
 * 
 * @type {!number}
 * @export
 */
API.Client.MintV3DTO.prototype.vid;

